packages/context/src/binding-filter.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Binding, BindingTag} from './binding';
+
+/**
+ * A function that filters bindings. It returns `true` to select a given
+ * binding.
+ */
+export type BindingFilter<ValueType = unknown> = (
+  binding: Readonly<Binding<ValueType>>,
+) => boolean;
+
+/**
+ * Create a binding filter for the tag pattern
+ * @param tagPattern
+ */
+export function bindingTagFilter(tagPattern: BindingTag | RegExp) {
+  let bindingFilter: BindingFilter;
+  if (typeof tagPattern === 'string' || tagPattern instanceof RegExp) {
+    const regexp =
+      typeof tagPattern === 'string'
+        ? wildcardToRegExp(tagPattern)
+        : tagPattern;
+    bindingFilter = b => Array.from(b.tagNames).some(t => regexp!.test(t));
+  } else {
+    bindingFilter = b => {
+      for (const t in tagPattern) {
+        // One tag name/value does not match
+        if (b.tagMap[t] !== tagPattern[t]) return false;
+      }
+      // All tag name/value pairs match
+      return true;
+    };
+  }
+  return bindingFilter;
+}
+
+/**
+ * Create a binding filter from key pattern
+ * @param keyPattern Binding key, wildcard, or regexp
+ */
+export function bindingKeyFilter(keyPattern?: string | RegExp) {
+  let filter: BindingFilter = binding => true;
```

Comment:
Similarly here:
 - don't use a mutable function, use `return` instead
 - use named function instead of `=>`